pilot/pkg/networking/core/v1alpha3/route/route.go

```diff
@@ -390,19 +390,17 @@ func translateRoute(push *model.PushContext, node *model.Proxy, in *networking.H
 			RetryPolicy: retry.ConvertPolicy(in.Retries),
 		}
 
+		// Configure timeouts specified by Virtual Service if they are provided, otherwise set it to defaults.
+		var d *duration.Duration
 		if in.Timeout != nil {
-			d := gogo.DurationToProtoDuration(in.Timeout)
-			// timeout
-			action.Timeout = d
-			action.MaxGrpcTimeout = d
+			d = gogo.DurationToProtoDuration(in.Timeout)
 		} else {
-			// if no timeout is specified, disable timeouts. This is easier
-			// to reason about than assuming some defaults.
-			d := ptypes.DurationProto(0 * time.Second)
-			action.Timeout = d
-			action.MaxGrpcTimeout = d
+			d = features.DefaultRequestTimeout()
 		}
 
+		action.Timeout = d
+		action.MaxGrpcTimeout = d
```

Comment:
do we want this to be the same? This field is a bit different

Reply:
Yes. It is bit different and is little confusing in Envoy it self. But IMO, it is better to keep the same because
- That is how originally it is there
- With out this, it is not possible to control grpc timeouts with  `grpc-timeout` header.

We can update virtual service doc to reflect what timeout field does for gRPC requests.
But definitely open here for other suggestions

Reply:
@howardjohn Thinking about this more, I think it may be a good idea to have two separate flags one per timeout (which is the current one) and another one for MaxGrpctimeout so that we do not have this confusion. Equating MaxGrpcTimeout to timeout does not seem right to me and it would break for users if their `grpc-timeout` header has value higher than what is confgured in the timout here.

WDYT? @hzxuzhonghu @rshriram  thoughts?

Reply:
I donot feel strong about adding another timeout env, because i think  in one route the request can be only one of http or grpc.  If we document this can be http timeout or grpc timeout, then it should be ok.

Reply:
>  If we document this can be http timeout or grpc timeout, then it should be ok

Please note this is not gRPC timeout. It is Max Grpc timeout. So if user sets it to 1s here and sets `grpc-timeout` in his gRPC client to 2s, it would not be honoured.

Reply:
> `Timeout → "grpc-timeout" TimeoutValue TimeoutUnit`

I  understand this value(1s) will override user client specified(2s).   And if there is no client side setting, this one takes effect, right?

Reply:
So you want to honor the user client setting, how can u solve it even you have a grpc specific env?

Reply:
> And if there is no client side setting, this one takes effect, right?

Yes. If there is no client side setting this value will be honoured for grpc request timeout.

> So you want to honor the user client setting, how can u solve it even you have a grpc specific env?
If we have a grpc specific env, If user knows all my clients set grpc-timeout header, user can set this to 0 (basically disable but allow the client to specify). If all of the clients does not specify grpc timeout, set this value to a higher value like 20s or some thing, and which ever clients set timeout, it would be honoured and rest will get this max timeout.  So there is a flexibility of choosing two values.

But not sure if it would be more confusing - I am torn between these two. 

But whatever others think is simpler to do, I can change.

Reply:
>But not sure if it would be more confusing - I am torn between these two.

Agree, this is not a once for all config.

Reply:
Ok. I have updated the comment and variable name and it continues to set the max_grpc_timeout. Let us see if others have any other opinions